dumpling/dumpling.go

```diff
@@ -0,0 +1,228 @@
+	} else {
+		select {
+		case <-ctx.Done():
+			isCanceled = true
+		default:
+		}
+	}
+
+	m.logger.Info("dump data finished", zap.Duration("cost time", time.Since(begin)))
+
+	pr <- pb.ProcessResult{
+		IsCanceled: isCanceled,
+		Errors:     errs,
+	}
+}
+
+// Close implements Unit.Close
+func (m *Dumpling) Close() {
+	if m.closed.Get() {
+		return
+	}
+	// do nothing, external will cancel the command (if running)
+	m.closed.Set(true)
+}
+
+// Pause implements Unit.Pause
+func (m *Dumpling) Pause() {
+	if m.closed.Get() {
+		m.logger.Warn("try to pause, but already closed")
+		return
+	}
+	// do nothing, external will cancel the command (if running)
+}
+
+// Resume implements Unit.Resume
+func (m *Dumpling) Resume(ctx context.Context, pr chan pb.ProcessResult) {
+	if m.closed.Get() {
+		m.logger.Warn("try to resume, but already closed")
+		return
+	}
+	// just call Process
+	m.Process(ctx, pr)
+}
+
+// Update implements Unit.Update
+func (m *Dumpling) Update(cfg *config.SubTaskConfig) error {
+	// not support update configuration now
+	return nil
+}
+
+// Status implements Unit.Status
+func (m *Dumpling) Status() interface{} {
+	// NOTE: try to add some status, like dumped file count
+	return &pb.DumpStatus{}
+}
+
+// Error implements Unit.Error
+func (m *Dumpling) Error() interface{} {
+	return &pb.DumpError{}
+}
+
+// Type implements Unit.Type
+func (m *Dumpling) Type() pb.UnitType {
+	return pb.UnitType_Dump
+}
+
+// IsFreshTask implements Unit.IsFreshTask
+func (m *Dumpling) IsFreshTask(ctx context.Context) (bool, error) {
+	return true, nil
+}
+
+// constructArgs constructs arguments for exec.Command
+func (m *Dumpling) constructArgs() (*export.Config, error) {
+	cfg := m.cfg
+	db := cfg.From
+
+	dumpConfig := export.DefaultConfig()
+	// block status addr because we already have it in DM, and if we enable it, may we need more ports for the process.
+	dumpConfig.StatusAddr = ""
+
+	dumpConfig.Host = db.Host
+	dumpConfig.Port = db.Port
+	dumpConfig.User = db.User
+	dumpConfig.Password = db.Password
+	dumpConfig.OutputDirPath = cfg.Dir // use LoaderConfig.Dir as output dir
+	dumpConfig.BlackWhiteList = export.BWListConf{
+		Mode: export.MySQLReplicationMode,
+		Rules: &export.MySQLReplicationConf{
+			Rules:         cfg.BWList,
+			CaseSensitive: cfg.CaseSensitive,
+		},
+	}
+	dumpConfig.EscapeBackslash = true
+	dumpConfig.Logger = m.logger.Logger
+
+	if cfg.Threads > 0 {
+		dumpConfig.Threads = cfg.Threads
+	}
+	if cfg.ChunkFilesize > 0 {
+		dumpConfig.FileSize = uint64(cfg.ChunkFilesize)
+	}
+	if cfg.StatementSize > 0 {
+		dumpConfig.StatementSize = cfg.StatementSize
+	}
+	if cfg.Rows > 0 {
+		dumpConfig.Rows = cfg.Rows
+	}
+	if len(cfg.Where) > 0 {
+		dumpConfig.Where = cfg.Where
+	}
+
+	var ret []string
+	if cfg.SkipTzUTC {
+		// TODO: support skip-tz-utc
+		ret = append(ret, "--skip-tz-utc")
+	}
+
+	extraArgs := strings.Fields(cfg.ExtraArgs)
+	if len(extraArgs) > 0 {
+		ret = append(ret, ParseArgLikeBash(extraArgs)...)
```

Comment:
do we need to parse the `extra-args`, like some config in the integration test, the user may set `extra-args: "--statement-size=100"`,  it may cause incompatible problem